internal/core/item_rendering.rs

```diff
@@ -158,6 +170,22 @@ pub trait ItemRenderer {
     #[cfg(feature = "std")]
     fn draw_path(&mut self, path: Pin<&Path>);
     fn draw_box_shadow(&mut self, box_shadow: Pin<&BoxShadow>);
+
+    // Apply the bounds of the Clip element, if enabled. The default implementation calls
+    // combine_clip, but the render may choose an alternate way of implementing the clip.
+    // For example the GL backend uses a layered rendering approach.
+    fn apply_clip(&mut self, clip_item: Pin<&Clip>, _self_rc: &ItemRc) -> RenderingResult {
```

Comment:
> This is not re-implemented in the PartialRenderer. That means the partial renderer can't work with renderers that use layers

Did you mean to leave that comment for `draw_opacity`  We don't need layer for apply_clip with most backend.

however, should that be called `visit_clip`  or something, because this doesn't apply the clip, combine_clip  does.
Same parallel between draw_opacity / apply_opacity   (maybe visit_ is better than draw_  in that case too.)

Reply:
`visit_` is a good prefix, I'll use that.